app/web/components/LanguagePickerSelect.tsx

```diff
@@ -0,0 +1,195 @@
+            backgroundColor: theme.palette.action.hover,
+          },
+        }}
+      >
+        <Stack
+          sx={{ width: "100%" }}
+          direction="row"
+          alignItems="center"
+          justifyContent="space-between"
+        >
+          <Stack direction="row">
+            <ListItemIcon>{renderFlag(flagCode)}</ListItemIcon>
+            <ListItemText
+              sx={{ color: "#666666", fontWeight: "bold", display: "inline" }}
+            >
+              {languageCode.toUpperCase()}
+            </ListItemText>
+          </Stack>
+          {/* if this menu item matches selected language, display a check mark */}
+          <div>
+            {language === languageCode && (
+              <CheckIcon fontSize="small" sx={{ color: "#00a69a" }} />
+            )}
+          </div>
+        </Stack>
+      </MenuItem>,
+    );
+  }
+
+  return (
+    <Box sx={{ minWidth: 60 }}>
+      <FormControl
+        variant="outlined"
+        displayMode={displayMode}
+        sx={{
+          // specialized sizing based on screen size
+          width:
+            displayMode === "round"
+              ? "fit-content"
+              : isMdOrWider
+                ? "241px"
+                : "100%",
+        }}
+      >
+        {" "}
+        {displayMode === "round" ? (
+          <MuiSelect
+            id="language-select"
+            value={language}
+            sx={{
+              borderRadius:
+                displayMode === "round" ? 999 : theme.shape.borderRadius,
+              "& .MuiSelect-icon": {
+                color: theme.palette.text.primary,
+                fontSize: "1.25rem",
+                top: "50%",
+                transform: "translateY(-50%)",
+                right: 10,
+              },
+            }}
+            onChange={handleChange}
+            // Use renderValue to display the selected language in collapsed state
+            renderValue={(selected: string) => {
+              const selectedDisplay = (
+                <Box
+                  sx={{
+                    display: "flex",
+                    alignItems: "center",
+                    gap: 1,
+                    pl: 1,
+                    color: "#666666",
+                    fontWeight: "bold",
+                  }}
+                >
+                  {renderFlag(LANGUAGE_MAP[selected].flagIconCode)}
+                  {selected.toUpperCase()}
+                </Box>
+              );
+              return displayMode === "round" ? (
+                selectedDisplay
+              ) : selected ? (
+                selectedDisplay
+              ) : (
+                <label className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1ttrm8x-MuiFormLabel-root-MuiInputLabel-root">
```

Comment:
What's up with all these MUI classes, looks a bit sus. What are you trying to accomplish here? I'm thinking there might be another way to go about it but not sure the goal from first glance.